query/query4_test.go

```diff
@@ -988,6 +989,246 @@ func TestCascadeSubQuery1(t *testing.T) {
+			  },
+			  {
+				"uid": "0x36",
+				"name": "D"
+			  },
+			  {
+				"uid": "0x3e9",
+				"name": "Bob"
+			  },
+			  {
+				"uid": "0x1001",
+				"name": "Badger"
+			  },
+			  {
+				"uid": "0x1001",
+				"name": "European badger"
+			  },
+			  {
+				"uid": "0x1001",
+				"name": "European badger barger European"
+			  },
+			  {
+				"uid": "0x1001",
+				"name": "Borsuk europejski"
+			  },
+			  {
+				"uid": "0x1001",
+				"name": "Europäischer Dachs"
+			  },
+			  {
+				"uid": "0x1001",
+				"name": "Барсук"
+			  },
+			  {
+				"uid": "0x1001",
+				"name": "Blaireau européen"
+			  },
+			  {
+				"uid": "0x13f2",
+				"name": "San Carlos"
+			  },
+			  {
+				"uid": "0x2717",
+				"name": "Elizabeth"
+			  }
+			]
+		  }
+	}`, js)
+}
+
+func TestHasEveryWithFirst(t *testing.T) {
+	query := `{
+		q(func:has(name), every:10, first: 2) {
+			 uid
+			 name
+		 }
+	 }`
+	js := processQueryNoErr(t, query)
+	require.JSONEq(t, `{
+		"data": {
+			"q": [
+			  {
+				"uid": "0x18",
+				"name": "Glenn Rhee"
+			  },
+			  {
+				"uid": "0x36",
+				"name": "D"
+			  }
+			]
+		  }
+	}`, js)
+}
+
+func TestHasEveryWithFirstOffset(t *testing.T) {
+	query := `{
+		q(func:has(name), every:10, first:2, offset:2) {
+			 uid
+			 name
+		 }
+	 }`
+	js := processQueryNoErr(t, query)
+	require.JSONEq(t, `{
+		"data": {
+			"q": [
+			  {
+				"uid": "0x3e9",
+				"name": "Bob"
+			  },
+			  {
+				"uid": "0x1001",
+				"name": "Badger"
+			  },
+			  {
+				"uid": "0x1001",
+				"name": "European badger"
+			  },
+			  {
+				"uid": "0x1001",
+				"name": "European badger barger European"
+			  },
+			  {
+				"uid": "0x1001",
+				"name": "Borsuk europejski"
+			  },
+			  {
+				"uid": "0x1001",
+				"name": "Europäischer Dachs"
+			  },
+			  {
+				"uid": "0x1001",
+				"name": "Барсук"
+			  },
+			  {
+				"uid": "0x1001",
+				"name": "Blaireau européen"
+			  }
+			]
+		  }
+	}`, js)
+}
+
+func TestHasEveryWithAfter(t *testing.T) {
+	query := `{
+		q(func:has(name), every:10, after:0x36) {
+			 uid
+			 name
+		 }
+	 }`
+	js := processQueryNoErr(t, query)
+	require.JSONEq(t, `{
+		"data": {
+			"q": [
+			  {
+				"uid": "0x3e9",
+				"name": "Bob"
+			  },
+			  {
+				"uid": "0x1001",
+				"name": "Badger"
+			  },
+			  {
+				"uid": "0x1001",
+				"name": "European badger"
+			  },
+			  {
+				"uid": "0x1001",
+				"name": "European badger barger European"
+			  },
+			  {
+				"uid": "0x1001",
+				"name": "Borsuk europejski"
+			  },
+			  {
+				"uid": "0x1001",
+				"name": "Europäischer Dachs"
+			  },
+			  {
+				"uid": "0x1001",
+				"name": "Барсук"
+			  },
+			  {
+				"uid": "0x1001",
+				"name": "Blaireau européen"
+			  },
+			  {
+				"uid": "0x13f2",
+				"name": "San Carlos"
+			  },
+			  {
+				"uid": "0x2717",
+				"name": "Elizabeth"
+			  }
+			]
+		  }
+	}`, js)
+}
+
+func TestHasEveryWithFirstAfter(t *testing.T) {
+	query := `{
+		q(func:has(name), every: 10, first:2, after: 0x36) {
```

Comment:
Shouldn't this return exactly 2 results?

Reply:
It returns exactly two uids, some have multiple `<name>` values.

Reply:
Did you modify the test now?

Reply:
Yes, I found another triple in the test data with predicate `<name>`, which changed the sequence of results, so I had to update the expectation: https://github.com/dgraph-io/dgraph/compare/52d436be0c4589528b9dffb1c593fa0a13b1f4de..a8e9948eb6f95dbcb879831a83cff4b465913172

I think that should be the right response, but my code produces some other result.

Reply:
Is there a way to unit-test `SubGraph.applyEvery` with some test `uidMatrix`?

Reply:
You were right, it should return only two results, as it selects the `name` property without language annotation per uid.